pkg/application/stateDiffUseCase.go

```diff
@@ -73,3 +78,48 @@ func (useCase *StateDiffUseCase) DetermineStateDiff(ctx context.Context, bluepri
 	// return it to signal that a repeated call to this function will not result in any progress.
 	return stateDiffError
 }
+
+func (useCase *StateDiffUseCase) collectClusterState(ctx context.Context, effectiveBlueprint domain.EffectiveBlueprint) (ecosystem.ClusterState, error) {
+	// TODO: collect cluster state in parallel (like for ecosystem health)
+	// load current dogus and components
+	installedDogus, doguErr := useCase.doguInstallationRepo.GetAll(ctx)
+	installedComponents, componentErr := useCase.componentInstallationRepo.GetAll(ctx)
+	// load current config
+	globalConfig, globalConfigErr := useCase.globalConfigRepo.GetAllByKey(ctx, effectiveBlueprint.Config.Global.GetGlobalConfigKeys())
+	doguConfig, doguConfigErr := useCase.doguConfigRepo.GetAllByKey(ctx, effectiveBlueprint.Config.GetDoguConfigKeys())
+	sensitiveDoguConfig, sensitiveConfigErr := useCase.sensitiveDoguConfigRepo.GetAllByKey(ctx, effectiveBlueprint.Config.GetSensitiveDoguConfigKeys())
+
+	joinedError := errors.Join(doguErr, componentErr, globalConfigErr, doguConfigErr, sensitiveConfigErr)
+
+	if joinedError != nil {
+		return ecosystem.ClusterState{}, fmt.Errorf("could not collect cluster state: %w", joinedError)
+	}
+
+	decryptedConfig, err := useCase.decryptSensitiveDoguConfig(ctx, sensitiveDoguConfig)
+	if err != nil {
+		return ecosystem.ClusterState{}, fmt.Errorf("could not decyrpt sensitive dogu config: %w", err)
+	}
+
+	return ecosystem.ClusterState{
+		InstalledDogus:               installedDogus,
+		InstalledComponents:          installedComponents,
+		GlobalConfig:                 globalConfig,
+		DoguConfig:                   doguConfig,
+		EncryptedDoguConfig:          sensitiveDoguConfig,
+		DecryptedSensitiveDoguConfig: decryptedConfig,
+	}, nil
+}
+
+func (useCase *StateDiffUseCase) decryptSensitiveDoguConfig(
+	ctx context.Context,
+	encryptedConfig map[common.SensitiveDoguConfigKey]*ecosystem.SensitiveDoguConfigEntry,
+) (map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue, error) {
```

Comment:
🚫 **[checkstyle]** <unparam> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>(*StateDiffUseCase).decryptSensitiveDoguConfig - result 1 (error) is always nil